docs/operating/linux.md

```diff
@@ -6,11 +6,129 @@ sidebar_position: 3
 
 ## systemd
 
-TigerBeetle includes an example
-[systemd unit](https://github.com/tigerbeetle/tigerbeetle/tree/main/tools/systemd/)
-for use with Linux systems that use systemd. The unit is configured to start a
-single-node cluster, so you may need to adjust it for other cluster
-configurations.
+The following includes an example systemd unit for running TigerBeetle with Linux systems that use systemd. The unit is configured to start a single-node cluster, so you may need to adjust it for other cluster configurations.
 
 See the [Quick Start](../quick-start.md) for an example of how to run a single-
 vs multi-node cluster.
+
+#### **tigerbeetle.service**
+```
+[Unit]
+Description=TigerBeetle Replica
+Documentation=https://docs.tigerbeetle.com/
+After=network-online.target
+Wants=network-online.target systemd-networkd-wait-online.service
+
+[Service]
+Environment=TIGERBEETLE_CACHE_GRID_SIZE=1GiB
+Environment=TIGERBEETLE_ADDRESSES=3001
+Environment=TIGERBEETLE_REPLICA_COUNT=1
+Environment=TIGERBEETLE_REPLICA_INDEX=0
+Environment=TIGERBEETLE_CLUSTER_ID=0
+Environment=TIGERBEETLE_DATA_FILE=%S/tigerbeetle/0_0.tigerbeetle
+
+DevicePolicy=closed
+DynamicUser=true
+LockPersonality=true
+ProtectClock=true
+ProtectControlGroups=true
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelLogs=true
+ProtectKernelModules=true
+ProtectKernelTunables=true
+ProtectProc=noaccess
+ProtectSystem=strict
+RestrictAddressFamilies=AF_INET AF_INET6
+RestrictNamespaces=true
+RestrictRealtime=true
+RestrictSUIDSGID=true
+
+StateDirectory=tigerbeetle
+StateDirectoryMode=700
+
+Type=exec
+ExecStartPre=/usr/local/bin/tigerbeetle-pre-start.sh
+ExecStart=/usr/local/bin/tigerbeetle start --cache-grid=${TIGERBEETLE_CACHE_GRID_SIZE} --addresses=${TIGERBEETLE_ADDRESSES} ${TIGERBEETLE_DATA_FILE}
+
+[Install]
+WantedBy=multi-user.target
+```
+
+#### **tigerbeetle-pre-start.sh**
+```bash
+#!/bin/sh
+set -eu
+
+if ! test -e "${TIGERBEETLE_DATA_FILE}"; then
+  /usr/local/bin/tigerbeetle format --cluster="${TIGERBEETLE_CLUSTER_ID}" --replica="${TIGERBEETLE_REPLICA_INDEX}" --replica-count="${TIGERBEETLE_REPLICA_COUNT}" "${TIGERBEETLE_DATA_FILE}"
+fi
+```
```

Comment:
We could remove `ExecStartPre=...` from the example and move this snippet to the "Pre-start script" section, making it sound more like a suggestion than the recommended way to create the replica data file.

IMO, this pre-start script is very opinionated.
The cluster operator might want to handle the storage provision and file creation in a different step.

Reply:
OK, I moved the script to the "Pre-start script" section.